aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogTable.java

```diff
@@ -343,28 +349,72 @@ public void testRenameTableFailsToCreateNewTable() {
   public void testRenameTableFailsToDeleteOldTable() {
     String namespace = createNamespace();
     String tableName = createTable(namespace);
-    // delete the old table metadata, so that drop old table will fail
+
+    // Create a spy on the Glue client
+    // Spy will simulate a time-out when trying to drop the old table during rename after new table
+    // already created
+    GlueClient glueClientSpy = spy(GLUE);
+    doAnswer(
+            invocation -> {
+              DeleteTableRequest originalRequest = invocation.getArgument(0);
+              if (originalRequest.name().equals(tableName)) {
+                DeleteTableRequest requestWithTimeout =
+                    originalRequest.toBuilder()
+                        // Inject short timeout threshold to the original request
+                        .overrideConfiguration(
+                            AwsRequestOverrideConfiguration.builder()
+                                .apiCallTimeout(Duration.ofMillis(1))
+                                .build())
+                        .build();
```

Comment:
this is some quite complicated logic to trigger a failure from Glue, is it possible to just change this line to throw an exception directly, or that might not hit the case we are looking to test? It also avoids taking dependency on `ApiCallTimeoutException.class` on L399/L402 which might change in future